plugins/item_tasks/server/rest.py

```diff
@@ -168,6 +168,30 @@ def _transformOutputs(self, outputs, token, job):
 
         return transformed
 
+    def _validateOutputParentType(self, outputId, parentType, outputSpec):
+        """
+        Checks if the output parent type is compatible with the output type.
+        """
+
+        # Find the corresponding output specification for the given outputID
+        output = None
```

Comment:
Python loop iteration variables persist in the scope outside, so you could use `out` in place of `output` below (or vice versa)